docs: fixes to ALTER EXTERNAL TABLE

mkiyama opened the pull request:
see issue https://github.com/greenplum-db/gpdb/issues/3338

@hlinnaka could you review this. 

HTML formatted output
http://docs-gpdb-review-staging.cfapps.io/510/ref_guide/sql_commands/ALTER_EXTERNAL_TABLE.html

dyozie commented on gpdb-doc/dita/ref_guide/sql_commands/ALTER_EXTERNAL_TABLE.xml:
```diff
@@ -1,41 +1,103 @@
+        ><title>Synopsis</title><codeblock id="sql_command_synopsis">ALTER EXTERNAL TABLE <varname>name</varname> <varname>action</varname> [, ... ]
 
-ALTER EXTERNAL TABLE <varname>name</varname> RENAME TO <varname>new_name</varname>
+where <varname>action</varname> is one of:
```
This sentence shouldn't be part of the codeblock.

mkiyama replied:
I put it in there because there is an issue with multiple code block sections in
command syntax in the syntax summary page. Only the first code block shows up.

I think I might have found a fix. We could just pull the summary page.

See the CREATE TABLE syntax.

http://docs-gpdb-review-staging.cfapps.io/510/ref_guide/GPReferenceSQLSummaryLOP.html

dyozie commented on gpdb-doc/dita/ref_guide/sql_commands/ALTER_EXTERNAL_TABLE.xml:
```diff
@@ -1,41 +1,103 @@
+        schema. To alter the owner, you must also be a direct or indirect member of the new owning
+        role, and that role must have <codeph>CREATE</codeph> privilege on the external table's
+        schema. A superuser has these privileges automatically.</p>
+      <p>Use the <codeph>ALTER TABLE</codeph> command to perform these actions.<ul
```
Think it might be clearest to say "to perform these actions **on an external table**:"

dyozie commented on gpdb-doc/dita/ref_guide/sql_commands/ALTER_EXTERNAL_TABLE.xml:
```diff
@@ -1,41 +1,103 @@
+          <pt>RESTRICT</pt>
+          <pd>Refuse to drop the column or constraint if there are any dependent objects. This is
+            the default behavior. </pd>
+        </plentry>
```
List of params should also include _expression_ for ```USING expression```

hlinnaka commented on gpdb-doc/dita/ref_guide/sql_commands/ALTER_EXTERNAL_TABLE.xml:
```diff
@@ -1,41 +1,103 @@
+          Greenplum Database. The <codeph>CASCADE</codeph> keyword is required if anything outside
+          the table depends on the column, such as a view that references the column. </li>
+        <li id="ak142648"><b>ALTER COLUMN TYPE</b> — Changes the data type of a table column. <p>The
+            optional <codeph>USING</codeph> clause specifies how to compute the new column value
```
The USING clause is quite silly for an external table, because the command won't actually rewrite the data. It won't be used for anything.

However, see https://github.com/greenplum-db/gpdb/issues/3356. We really should fix that in code, and after that, probably throw an error if USING is specified with ALTER EXTERNAL TABLE.

mkiyama replied:
@hlinnaka Is there an issue or story to remove the USING clasue

hlinnaka commented on gpdb-doc/dita/ref_guide/sql_commands/ALTER_EXTERNAL_TABLE.xml:
```diff
@@ -1,41 +1,103 @@
+          user. </li>
+      </ul>
+      <p>You must own the external table to use <codeph>ALTER EXTERNAL TABLE</codeph>. To change the
+        schema of an external table, you must also have <codeph>CREATE</codeph> privilege on the new
```
The discussion on changing the schema seems misplaced, since `ALTER TABLE SET SCHEMA` is listed only after this.

mkiyama commented:
@hlinnaka @dyozie 
updated - I think I fixed most things.
Need to know whether or not to add that USING clause is deprecated.

HTML formatted output
http://docs-gpdb-review-staging.cfapps.io/510/ref_guide/sql_commands/ALTER_EXTERNAL_TABLE.html

Eulerizeit commented:
It isn't deprecated. I do think it makes sense to have that conversation though. 

@ivannovick

hlinnaka commented:
On 10/04/2017 09:33 PM, Robert Eckhardt wrote:
> It isn't deprecated. I do think it makes sense to have that conversation though.

We should fix the code, so that ALTER EXTERNAL TABLE doesn't require a 
dummy USING. Per https://github.com/greenplum-db/gpdb/issues/3356. And 
then we should remove the USING clause from ALTER EXTERNAL TABLE 
altogether. And then document that.

- Heikki
